### ISSUE TITLE
PyLint false positive PLW0602 when global is a dict

### ISSUE THREAD

**Issue by NicolasSassine** [astral-sh/ruff#27205]
### Summary

Hello,

Modifying a global dictionary does not clear away the ruff PLW0602 error.
Example output of ruff check on minimal code:
```
PLW0602 Using global for `TEST_VARIABLE` but no assignment is done
 --> tmp.py:4:12
  |
3 | def test_my_variable():
4 |     global TEST_VARIABLE
  |            ^^^^^^^^^^^^^
5 |     TEST_VARIABLE['hello'] = 'world'
6 |     return TEST_VARIABLE
  |
```

The definition of `TEST_VARIABLE` is done on line 1, as `TEST_VARIABLE = {}`.
I can also confirm that doing TEST_VARIABLE.pop() doesn't clear the error either.

### Version

ruff 0.16.10

**Comment by Avasam** [comment]
That's accurate. You're mutating, not assigning. So you don't need to declare the global scope.

**Comment by NicolasSassine** [comment]
> That's accurate. You're mutating, not assigning. So you don't need to declare the global scope.

Thanks for pointing this out! It makes sense, I'll close the issue.
